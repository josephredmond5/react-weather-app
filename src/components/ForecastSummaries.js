import React from "react";
import PropTypes from "prop-types";
import ForecastSummary from "./ForecastSummary";

function ForecastSummaries({ forecasts }) {
  return (
    <div>
      {forecasts.map((forecast) => (
        <ForecastSummary key={forecast.date} {...forecast} />
      ))}
    </div>
  );
}

ForecastSummaries.propTypes = {
  forecasts: PropTypes.arrayOf(
    PropTypes.shape({
      date: PropTypes.string.isRequired,
      description: PropTypes.string.isRequired,
      icon: PropTypes.string.isRequired,
      temperature: PropTypes.shape({
        max: PropTypes.number,
        min: PropTypes.number,
      }).isRequired,
    })
  ).isRequired,
};

export default ForecastSummaries;
