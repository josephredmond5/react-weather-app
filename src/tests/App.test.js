import React from "react";
import { render } from "@testing-library/react";
import ForecastSummaries from "../components/ForecastSummaries";

describe("ForecastSummaries", () => {
  const validProps = [
    {
      date: "2022-04-04",
      description: "Stub description 1",
      icon: "stubIcon1",
      temperature: {
        max: 22,
        min: 12,
      },
    },
    {
      date: "2022-04-05",
      description: "Stub description 2",
      icon: "stubIcon2",
      temperature: {
        max: 24,
        min: 14,
      },
    },
  ];
  

  it("renders correctly", () => {
    const { asFragment } = render(<ForecastSummaries forecasts={validProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
