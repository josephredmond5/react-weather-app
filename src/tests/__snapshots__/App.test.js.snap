// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummaries renders correctly 1`] = `
<DocumentFragment>
  <div>
    <div
      class="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        2022-04-04
      </div>
      <div
        class="forecast-summary__icon"
        data-testid="forecast-icon"
      >
        <img
          alt="Stub description 1"
          src="stubIcon1"
        />
      </div>
      <div
        class="forecast-summary__temperature"
      >
        22°C
      </div>
      <div
        class="forecast-summary__description"
      >
        Stub description 1
      </div>
    </div>
    <div
      class="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        2022-04-05
      </div>
      <div
        class="forecast-summary__icon"
        data-testid="forecast-icon"
      >
        <img
          alt="Stub description 2"
          src="stubIcon2"
        />
      </div>
      <div
        class="forecast-summary__temperature"
      >
        24°C
      </div>
      <div
        class="forecast-summary__description"
      >
        Stub description 2
      </div>
    </div>
  </div>
</DocumentFragment>
`;
